code/mission/missionparse.cpp

```diff
@@ -4729,15 +4740,44 @@ void parse_wing(mission *pm)
 	// Goober5000 - wing creation stuff moved to post_process_ships_wings
 }
 
-void parse_wings(mission *pm)
+void parse_wings(mission* pm)
 {
+	// reset this for the missing wing bug.
+	for (int i = 0; i < MAX_SQUADRON_WINGS; i++) {
+		Squadron_wing_names_found[i] = false;
+	}
+
 	required_string("#Wings");
 	while (required_string_either("#Events", "$Name:"))
 	{
 		Assert(Num_wings < MAX_WINGS);
 		parse_wing(pm);
 		Num_wings++;
 	}
+
+	bool found = false;
+
+	// now see if we found the missing wing.  We're looking for a wing that is there after a wing that is not.
+	// for now TvT mission do not have enough player wings to be affected by this bug.
```

Comment:
I would add an Assertion here to check that the number of TvT wings is under the bug threshold.  In the event that someone bumps TvT wings in the future, he may not think to check here.

Reply:
Smart. Thanks for the review!

Reply:
See what you think of that assert.

Reply:
Looks good!